### PR TITLE
Java 8 Migration: remove uneccessary if statement

### PR DESCRIPTION
**Pull request by kezhenxu94**
Motivation:

Remove unnecessary if statement

Modification:

Remove unnecessary if statement

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@kezhenxu94 there is a compilation failure... maybe because of a bad merge

File: common/src/main/java/io/netty/util/internal/ThrowableUtil.java [hunk 1]
```diff
@@ -54,16 +54,9 @@ public static String stackTraceToString(Throwable cause) {
...
-    public static boolean haveSuppressed() {
-        return PlatformDependent.javaVersion() >= 7;
-    }
-
     @SuppressJava6Requirement(reason = "Throwable addSuppressed is only available for >= 7. Has check for < 7.")
     public static void addSuppressed(Throwable target, Throwable suppressed) {
-        if (!haveSuppressed()) {
-            return;
-        }
-        target.addSuppressed(suppressed);
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/ThrowableUtil.java]: @kezhenxu94 I think this change is not correct and it should always call `target.addSuppressed(...)`. Please fix

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java [hunk 2]
```diff
@@ -341,6 +332,13 @@ public static void throwException(Throwable t) {
      * Creates a new fastest {@link ConcurrentMap} implementation for the current platform.
      */
     public static <K, V> ConcurrentMap<K, V> newConcurrentHashMap() {
+        return new ConcurrentHashMap<K, V>();
+    }
+
+    /**
+     * Creates a new fastest {@link LongCounter} implementation for the current platform.
+     */
+    public static LongCounter newLongCounter() {
```
Comment (njhill) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: Something is wrong here....

**Comment by kezhenxu94** [comment]
thanks for reviewing, will fix them later

**Comment by kezhenxu94** [comment]
@normanmaurer sure I will

**Comment by kezhenxu94** [comment]
@normanmaurer Done

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by kezhenxu94** [comment]
@normanmaurer Sorry, my local environment seems to be broken and make code style violated with checkstyle rules, I'll need some time to fix it.

**Comment by kezhenxu94** [comment]
@normanmaurer I've validate in my local environment and validation passed, it seems that the CI failed because of a unrelated issue: 
image
mind having a look and re-run the test if possible?

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: common/src/main/java/io/netty/util/internal/PlatformDependent.java [hunk 3]
```diff
@@ -1164,27 +1160,27 @@ static int hashCodeAsciiSafe(byte[] bytes, int startPos, int length) {
             hash = PlatformDependent0.hashCodeAsciiCompute(getLongSafe(bytes, i), hash);
         }
         switch(remainingBytes) {
-        case 7:
-            return ((hash * HASH_CODE_C1 + hashCodeAsciiSanitize(bytes[startPos]))
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/internal/PlatformDependent.java]: @kezhenxu94 can you please revert all the "formatting-only" changes ?

Comment (kezhenxu94) [reply common/src/main/java/io/netty/util/internal/PlatformDependent.java]: DONE

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@kezhenxu94 thanks

**Pull request merged**